tensorflow_datasets/core/utils/py_utils.py

```diff
@@ -275,3 +275,9 @@ def rgetattr(obj, attr, *args):
   def _getattr(obj, attr):
     return getattr(obj, attr, *args)
   return functools.reduce(_getattr, [obj] + attr.split("."))
+
+
+def free_disk_size(directory):
```

Comment:
(copied from earlier)

Overall we need to think about doing this in a way that's robust to:
* OS: Linux, Windows
* Machines with multiple disks (i.e. we need to use the one that data_dir is on)
* Cloud filesystems (i.e. if the data_dir were set to `gs://...`; in that case we want to skip this check)

Reply:
Sorry I just missed it. I am fixed immediately.

Reply:
Fixed with using `_data_dir_root`.